tests/test_flow_handler.py

```diff
@@ -23,3 +27,294 @@ def test_is_supported_flow_without_ts(
     flow.type_ = flow_type
     flow_handler = ModuleFactory().create_flow_handler_obj(flow, mock_db)
     assert flow_handler.is_supported_flow() == expected_val
+
+
+def test_handle_dns():
+    mock_db = Mock()
+    flow = Mock()
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.twid = "timewindow_id"
+    flow_handler.profileid = "profile_id"
+    flow_handler.handle_dns()
+
+    mock_db.add_out_dns.assert_called_with(
+        flow_handler.profileid,
+        flow_handler.twid,
+    )
+    mock_db.add_altflow.assert_called_with(
+        flow, flow_handler.profileid, flow_handler.twid, "benign"
+    )
+
+
+# testing handle_ftp
+def test_handle_ftp():
+    mock_db = Mock()
+    flow = Mock()
+    flow.used_port = 21  # Assuming FTP typically uses port 21
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.profileid = "profile_id"
+    flow_handler.twid = "timewindow_id"
+    flow_handler.handle_ftp()
+
+    mock_db.set_ftp_port.assert_called_with(21)
+    mock_db.add_altflow.assert_called_with(
+        flow, flow_handler.profileid, flow_handler.twid, "benign"
+    )
+
+
+def test_handle_http():
+    mock_db = Mock()
+    flow = Mock()
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.profileid = "profile_id"
+    flow_handler.twid = "timewindow_id"
+    flow_handler.handle_http()
+
+    mock_db.add_out_http.assert_called_with(
+        flow_handler.profileid, flow_handler.twid, flow
+    )
+    mock_db.add_altflow.assert_called_with(
+        flow, flow_handler.profileid, flow_handler.twid, "benign"
+    )
+
+
+# testing handle_ssl
+def test_handle_ssl(flow, mock_db):
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.profileid = "profile_id"
+    flow_handler.twid = "timewindow_id"
+    flow_handler.handle_ssl()
+
+    mock_db.add_out_ssl.assert_called_with(
+        flow_handler.profileid, flow_handler.twid, flow
+    )
+    mock_db.add_altflow.assert_called_with(
+        flow, flow_handler.profileid, flow_handler.twid, "benign"
+    )
+
+
+# testing handle_ssh
+def test_handle_ssh(flow, mock_db):
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.profileid = "profile_id"
+    flow_handler.twid = "timewindow_id"
+    flow_handler.handle_ssh()
+
+    mock_db.add_out_ssh.assert_called_with(
+        flow_handler.profileid, flow_handler.twid, flow
+    )
+    mock_db.add_altflow.assert_called_with(
+        flow, flow_handler.profileid, flow_handler.twid, "benign"
+    )
+
+
+# testing handle_weird
+def test_handle_weird(flow, mock_db):
+    flow_handler = FlowHandler(mock_db, None, flow)
+    flow_handler.profileid = "profile_id"
+    flow_handler.twid = "timewindow_id"
+    flow_handler.handle_weird()
+
+    mock_db.publish.assert_called_with("new_weird", ANY)
```

Comment:
Hello @Sekhar-Kumar-Dash 
why are you using ANY? insead of passing the flow as a dict? i think that would be better

Reply:
Hey @AlyaGomaa  just fixed it please take a look